src/chart/Column.js

```diff
@@ -31,46 +31,62 @@
 
         this._palette = this._palette.switch(this.paletteID());
 
-        var column = this.svgData.selectAll(".columnRect")
+        var columnScale = d3.scale.ordinal();
+
+        columnScale.domain(context._columns.filter(function (d, idx) { return idx > 0; })).rangeRoundBands([0, context.dataScale.rangeBand()]);
+
+        var column = this.svgData.selectAll(".dataRow")
             .data(this.data())
         ;
 
-        var title = column
-          .enter().append("rect")
-            .attr("class", "columnRect")
-            .on("click", function (d) {
-                context.click(context.rowToObj(d), context._columns[1]);
-            })
-            .append("title")
+        column.enter().append("g")
+            .attr("class", "dataRow")
         ;
 
-        if (this.orientation() === "horizontal") {
-            column.transition()
-                .attr("class", "columnRect")
-                .attr("x", function (d) { return context.dataScale(d[0]); })
-                .attr("width", this.dataScale.rangeBand ? this.dataScale.rangeBand() : 10)
-                .attr("y", function (d) { return context.valueScale(d[1]); })
-                .attr("height", function (d) { return height - context.valueScale(d[1]); })
-                .style("fill", function (d) { return context._palette(d[0]); })
-            ;
-        } else {
-            column.transition()
-                .attr("class", "columnRect")
-                .attr("x", function (d) { return 0; })
-                .attr("height", this.dataScale.rangeBand ? this.dataScale.rangeBand() : 10)
-                .attr("y", function (d) { return context.dataScale(d[0]); })
-                .attr("width", function (d) { return context.valueScale(d[1]); })
-                .style("fill", function (d) { return context._palette(d[0]); })
-            ;
-        }
-
-        title
-            .text(function (d) { return d[0] + " (" + d[1] + ")"; })
-        ;
+        column
+            .each(function (dataRow, i) {
+                var element = d3.select(this);
+
+                var columnRect = element.selectAll("rect").data(dataRow.filter(function(d, i) {return i > 0;}));
+
+                var title = columnRect
+                  .enter().append("rect")
+                    .attr("class", "columnRect")
+                    .on("click", function (d, idx) {
+                        context.click(context.rowToObj(idx), dataRow + " " + context._columns[idx + 1]);
```

Comment:
Click is not passing the correct data, it should be the entire dataRow and Column Name.